File: 0788-rotated-digits/0788-rotated-digits.cpp
class Solution {
public:
    int rotatedDigits(int n) {
        int cnt = 0;
        for (int num = 1; num <= n; num++) {
            int check = num;
            bool valid = true;
            bool changed = false;
            while (check > 0 && valid) {
                int digit = check % 10;
                if (digit == 3 || digit == 4 || digit == 7) {
                    valid = false;
                } else if (digit == 2 || digit == 5 || digit == 6 || digit == 9) {
                    changed = true;
                }
                check /= 10;
            }
            if (valid && changed) 
                cnt++;
        }
        return cnt;
    }
};

File: 0788-rotated-digits/0788-rotated-digits.py
class Solution:
    def rotatedDigits(self, n: int) -> int:
        cnt = 0
        for num in range(1, n + 1):
            check = num
            valid = True
            changed = False
            while check > 0 and valid:
                digit = check  % 10
                if digit in (3, 4, 7):
                    valid = False
                elif digit in (2, 5, 6, 9):
                    changed = True
                check //= 10
            if valid and changed:
                cnt += 1
        return cnt